pkg/common/aws/iam.go

```diff
@@ -0,0 +1,209 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+)
+
+var (
+	rolesubstr     = "osde2e"
+	providersubstr = "cloudfront"
+)
+
+func (CcsAwsSession *ccsAwsSession) CleanupOpenIDConnectProviders(olderthan time.Duration, dryrun bool) error {
+	err := CcsAwsSession.GetAWSSessions()
+	if err != nil {
+		return err
+	}
+
+	input := &iam.ListOpenIDConnectProvidersInput{}
+	result, err := CcsAwsSession.iam.ListOpenIDConnectProviders(input)
+	if err != nil {
+		return err
+	}
+
+	for _, provider := range result.OpenIDConnectProviderList {
+		// Create GetOpenIDConnectProviderInput struct
+		input := &iam.GetOpenIDConnectProviderInput{
+			OpenIDConnectProviderArn: provider.Arn,
+		}
+
+		// Get the provider
+		result, err := CcsAwsSession.iam.GetOpenIDConnectProvider(input)
+		if err != nil {
+			return err
+		}
+
+		// If provider name contains "cloudfront" and is older than given days, delete it
+		if strings.Contains(*result.Url, providersubstr) && time.Since(*result.CreateDate) > olderthan {
```

Comment:
What is the reasoning for waiting? If the cluster is deleted, I would assume this resource is no longer valuable. I don't check for dates when I do my cleanup.

Reply:
Oh, I guess that makes a bit of sense. I would prefer the implementation of this to filter out any cluster resources that are running but we can go with this.